sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVDataSource.scala

```diff
@@ -64,7 +64,9 @@ abstract class CSVDataSource extends Serializable {
       inputPaths: Seq[FileStatus],
       parsedOptions: CSVOptions): Option[StructType] = {
     if (inputPaths.nonEmpty) {
-      Some(infer(sparkSession, inputPaths, parsedOptions))
+      // Remove "path" option so that it is not added to the given `inputPaths`.
```

Comment:
Can we do it at the caller side of `FileFormat.inferSchema`? Then we don't need to change various file source implementations.

Reply:
But it's not going to fix v2 datasources since `inferSchema` is called from `FileTable`.

Reply:
And `FileTable.inferSchema` doesn't take in options.

Reply:
For v1, I can fix it at the caller side of `FileFormat.inferSchema` to cover all datasources. And for v2, I can fix it at `CSVTable.inferSchema` and `JSONTable.inferSchema`; the other built-in datasources don't seem to be affected. WDYT?

Reply:
Good point on v2.

I think in v2, the `FileTable.options` should not contain `path/paths` anymore, as we already extracted the paths from the options. Can we do some refactor to also clear the path options in a higher level for v2 as well?